### PR TITLE
Add hazelcast and hazelcast-enterprise Helm Charts

### PR DESCRIPTION
**Pull request by leszko**
Two Helm Charts added:
- Hazelcast = Hazelcast OSS from official repo + Management Center
- Hazelcast Enterprise = Hazelcast (with licenseKey value and default image values changed)

I kept exactly the same conventions as the official Helm Chart repo, because we will have to keep in sync our repo with the official one.

### CODE REVIEW

File: stable/hazelcast-enterprise/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: hazelcast-enterprise
+version: 1.0.0
+appVersion: "3.10.2"
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/Chart.yaml]: maybe update to 3.10.3?

File: stable/hazelcast-enterprise/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
+# Hazelcast Enterprise
+
+[Hazelcast IMDG Enterprise](https://hazelcast.com/products/enterprise/) is the most widely used in-memory data grid with hundreds of thousands of installed clusters around the world. It offers caching solutions ensuring that data is in the right place when it’s needed for optimal performance.
+
+## Quick Start
+
+```bash
+$ helm repo add hazelcast <hazelcast_repo_url>
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/README.md]: I think we should replace it with the exact URL when we decide where to host chart files.

File: stable/hazelcast-enterprise/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following table lists the configurable parameters of the Hazelcast chart and their default values.
+
+| Parameter                                  | Description                                                                                                    | Default                                              |
+|--------------------------------------------|----------------------------------------------------------------------------------------------------------------|------------------------------------------------------|
+| `image.repository`                         | Hazelcast Image name                                                                                           | `hazelcast/hazelcast-enterprise-kubernetes`          |
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/README.md]: Didn't we make base hazelcast docker image k8s compatible?

File: stable/hazelcast-enterprise/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+| `serviceAccount.create`                    | Enable installing Service Account                                                                              | `true`                                               |
+| `serviceAccount.name`                      | Name of Service Account, if not set, the name is generated using the fullname template                         | `nil`                                                |
+| `mancenter.enabled`                        | Turn on and off Management Center application                                                                  | `true`                                               |
+| `mancenter.image.repository`               | Hazelcast Management Center Image name                                                                         | `hazelcast/management-center`                        |
+| `mancenter.image.tag`                      | Hazelcast Management Center Image tag                                                                          | `{VERSION}`                                          |
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/README.md]: We need to mention somewhere that VERSION will be the same as hazelcast version by default. Any custom version needs to follow compatibility rule... which is MC version must be the same or one minor release higher than hazelcast version.

File: stable/hazelcast-enterprise/templates/config.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+kind: ConfigMap
+metadata:
+  name: {{ template "hazelcast.fullname" . }}-configuration
+  labels:
+    app: {{ template "hazelcast.name" . }}
+    chart: {{ template "hazelcast.chart" . }}
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+data:
+{{- range $key, $val := .Values.hazelcast.configurationFiles }}
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/config.yaml]: What do we have exactly here?

File: stable/hazelcast/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+version: 1.0.0
+appVersion: "3.10.2"
+description: Hazelcast IMDG is the most widely used in-memory data grid with hundreds of thousands of installed clusters around the world. It offers caching solutions ensuring that data is in the right place when it’s needed for optimal performance.
+keywords:
+- hazelcast
+- keyvalue
+- in-memory
+- database
+- caching
+home: http://hazelcast.com/
```
Comment (mesutcelik) [inline stable/hazelcast/Chart.yaml]: .org?

File: stable/hazelcast/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+icon: http://hazelcast.com/images/logos/HazelcastLogo-Blue_Dark_Icon_200px.png
```
Comment (mesutcelik) [inline stable/hazelcast/Chart.yaml]: orange icon?
http://hazelcast.com/images/logos/Hazelcast-IMDG-Logo-Orange_Dark_Icon_200px.png

File: stable/hazelcast-enterprise/README.md (hunk 2 above)
Comment (leszko) [reply stable/hazelcast-enterprise/README.md]: yeah, agree. I have it on my list. However, I'd keep it as it in this PR.

File: stable/hazelcast-enterprise/templates/config.yaml (hunk 5 above)
Comment (leszko) [reply stable/hazelcast-enterprise/templates/config.yaml]: For each configuration files mentioned in `configurationFiles` in `values.yaml` put a config map here. One could argue that we will not need any other configuration file than `hazelcast.xml`, but it's simple this way and I'd like to keep it consistent with what I've sent in PR to the official Helm Chart repo.

File: stable/hazelcast/Chart.yaml (hunk 7 above)
Comment (leszko) [reply stable/hazelcast/Chart.yaml]: Are there any guidelines on which icon to use when? I'd keep it the same as what we've sent to the official Helm Chart.

File: stable/hazelcast-enterprise/Chart.yaml (hunk 1 above)
Comment (leszko) [reply stable/hazelcast-enterprise/Chart.yaml]: Updated.

File: stable/hazelcast-enterprise/README.md (hunk 3 above)
Comment (leszko) [reply stable/hazelcast-enterprise/README.md]: Yeah, up from `3.10.3` we can use `hazelcast/hazelcast-enterprise`. Updated.

File: stable/hazelcast-enterprise/README.md (hunk 4 above)
Comment (leszko) [reply stable/hazelcast-enterprise/README.md]: Right, I've added a comment. PTAL.

File: stable/hazelcast-enterprise/templates/config.yaml (hunk 5 above)
Comment (mesutcelik) [reply stable/hazelcast-enterprise/templates/config.yaml]: Can we pass a configuration file to helm chart or it is just a ConfigMap created with specified name but empty content?

File: stable/hazelcast/Chart.yaml (hunk 6 above)
Comment (mesutcelik) [reply stable/hazelcast/Chart.yaml]: We do promote .org in Hazelcast OSS so I thought that is the way to go.

File: stable/hazelcast/Chart.yaml (hunk 7 above)
Comment (mesutcelik) [reply stable/hazelcast/Chart.yaml]: https://hazelcast.com/brand-assets/

These are the explanation of different tabs.
Hazelcast --> Company Assets
Hazelcast IMDG Blue --> Hazelcast Enterprise
Hazelcast IMDG Orange --> Hazelcast Opensource
Hazelcast Jet (Orange) --> Hazelcast Jet

Comment (leszko) [reply stable/hazelcast/Chart.yaml]: Thanks for explanation. I've updated both logos.

File: stable/hazelcast/Chart.yaml (hunk 6 above)
Comment (leszko) [reply stable/hazelcast/Chart.yaml]: ok, changed to `.org`. I'll changed it also in the official repo (but they stopped reviewing it again (!))

**Comment by leszko** [comment]
@mesutcelik , I've applied the changed. PTAL.

**Comment by leszko** [comment]
Btw. I needed to move back the version `hazelcast:3.10.3` to `hazelcast-kubernetes:3.10.2`, because the MC version should not be lower than HZ.

**Comment by emre-aydin** [comment]
@mesutcelik @leszko it shouldn't be a problem.

**Comment by emre-aydin** [comment]
> Btw. I needed to move back the version hazelcast:3.10.3 to hazelcast-kubernetes:3.10.2, because the MC version should not be lower than HZ.

@leszko the only requirement for MC is to have the same or plus one minor version of HZ. Patch versions don't matter.

**Comment by leszko** [comment]
@emre-aydin @mesutcelik Thanks for explanation. Updated to `3.10.3`.

**Pull request merged**